### PR TITLE
feature/ST-495-allow-placeholder-to-be-overridden-by-merchant-configuration

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/core/config/ConfigResolver.ts [hunk 1]
```diff
@@ -67,6 +66,8 @@ export class ConfigResolver {
       jwt: config.jwt !== undefined ? config.jwt : '',
       livestatus: config.livestatus !== undefined ? config.livestatus : 0,
       origin: config.origin !== undefined ? config.origin : window.location.origin,
+      placeholders:
```
Comment (wirus15) [inline src/core/config/ConfigResolver.ts]:
```
placeholders: config.placeholders || { pan: '', expirydate: '', securitycode: '' },
```
shorter 😉

File: test/components/card-number/CardNumber.spec.ts [hunk 2]
```diff
@@ -494,16 +497,18 @@ function cardNumberFixture() {
   const html =
     '<form id="st-card-number" class="card-number" novalidate=""><label id="st-card-number-label" for="st-card-number-input" class="card-number__label card-number__label--required">Card number</label><input id="st-card-number-input" class="card-number__input" type="text" autocomplete="off" required="" data-luhn-check="true" maxlength="NaN" minlength="19"><p id="st-card-number-message" class="card-number__message"></p></form>';
   document.body.innerHTML = html;
-  let cardNumberInstance = new CardNumber();
+  let configService: ConfigService;
+  configService = mock(ConfigService);
+  const cardNumberInstance = new CardNumber(configService);
```
Comment (wirus15) [inline test/components/card-number/CardNumber.spec.ts]: when you use the mock in some code you should create an instance of it:
```typescript
const cardNumberInstance = new CardNumber(instance(configService));

File: src/core/config/ConfigResolver.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/core/config/ConfigResolver.ts]: 🤦‍♂  I always forget about this, thx 👍

File: test/components/card-number/CardNumber.spec.ts (hunk 2 above)
Comment (pgs-gmartowski) [reply test/components/card-number/CardNumber.spec.ts]: is it a Jest function ? I couldn't find this :/

Comment (pgs-gmartowski) [reply test/components/card-number/CardNumber.spec.ts]: done

File: src/core/config/ConfigResolver.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/core/config/ConfigResolver.ts]: done

**Pull request closed without merging**